src/prefect/server/database/configurations.py

```diff
@@ -337,6 +347,12 @@ async def engine(self) -> AsyncEngine:
             if self.timeout is not None:
                 kwargs["connect_args"] = dict(timeout=self.timeout)
 
+            if self.sqlalchemy_pool_size is not None:
+                kwargs["pool_size"] = self.sqlalchemy_pool_size
+
+            if self.sqlalchemy_max_overflow is not None:
+                kwargs["max_overflow"] = self.sqlalchemy_max_overflow
+
```

Comment:
Also, I added these kwargs into our connection pools for non-memory SQLite as well; I couldn't find any reason why we should avoid configuring the pool for a SQLite DB.